src/state/opamFormatUpgrade.ml

```diff
@@ -1066,25 +1066,30 @@ let downgrade_2_1_switches root conf =
     (OpamFile.Config.installed_switches conf);
   conf
 
-let from_2_1_alpha_to_2_1_alpha2 root conf =
-  downgrade_2_1_switches root conf
+let from_2_1_alpha_to_2_1_alpha2 root ~on_the_fly:_ conf =
+  downgrade_2_1_switches root conf, Some `Switch
 
-let from_2_1_alpha2_to_v2_1_rc root conf =
-  downgrade_2_1_switches root conf
+let from_2_1_alpha2_to_v2_1_rc root ~on_the_fly:_ conf =
+  downgrade_2_1_switches root conf, Some `Switch
 
-let from_2_1_rc_to_v2_1 _ conf = conf
+let from_2_1_rc_to_v2_1 _ ~on_the_fly:_ conf = conf, None
 
-let from_2_0_to_v2_1 _ conf =
+let from_2_0_to_v2_1 _ ~on_the_fly conf =
   (* In opam < 2.1 "jobs" was set during initialisation
      This creates problems when upgrading from opam 2.0 as it
      sets the job count for good even if the CPU is replaced.
      See https://github.com/ocurrent/ocaml-dockerfile/pull/92 *)
   let info_jobs_changed ~prev_jobs =
-    OpamConsole.note
-      "The 'jobs' option was reset, its value was %d and its new value \
-      will vary according to the current number of cores on your machine. \
-       If it really was intended, you can set it again using:\n    \
-       opam option jobs=%d --global" prev_jobs prev_jobs;
+    if on_the_fly then
+      OpamConsole.note
+        "The 'jobs' option was reset for this launch, its new value will vary \
+         according to the current number of cores on your machine."
+    else
+      OpamConsole.note
+        "The 'jobs' option was reset, its value was %d and its new value \
+         will vary according to the current number of cores on your machine. \
+         If it really was intended, you can set it again using:\n    \
+         opam option jobs=%d --global" prev_jobs prev_jobs;
```

Comment:
@kit-ty-kate @dra27 we need to have an on-the-fly message and more complete written upgrade on. Any ideas about the on-the-fly one? It needs to be simple as it can be shown multiple times before user launches a command that takes a write lock on global state.

Reply:
After discussion, removing it.
```suggestion
    if not on_the_fly then
      OpamConsole.note
        "The 'jobs' option was reset, its value was %d and its new value \
         will vary according to the current number of cores on your machine. \
         If it really was intended, you can set it again using:\n    \
         opam option jobs=%d --global" prev_jobs prev_jobs;
```